proposals/lip-0071.md

```diff
@@ -269,7 +272,36 @@ The default reward of the block as a 64-bit unsigned integer.
 
 ### Endpoints for Off-Chain Services
 
-TBD.
+#### getExpectedValidatorRewards
+
+This function is used to estimate the expected rewards for a validator assuming the current BFT weight distribution. It provides an estimate of rewards for one block, one day, one month and one year.
+
+##### Returns
+
+* `reward:` amount of block reward for a validator.
+
+##### Execution
+
+```python
+def getExpectedValidatorRewards(validatorAddress: Address) -> tuple[uint64, uint64, uint64, uint64]:
+    validatorParams = Validators.getValidatorParams()
+    totalBftWeight = sum(validator.bftWeight for validator in validatorParams.validators)
+
+    let validator be the item of validatorParams.validators with item.address == validatorAddress
+    if not validator:
+        return (0, 0, 0, 0) 
+
+    if validator.bftWeight > 0:
+        blockReward = minimumRewardActiveValidators + (validator.bftWeight * stakeRewardActiveValidators)//totalBftWeight 
+    else:
+        blockReward = defaultRewardStandByValidators
+
+    dailyReward =   86400 // (Pos.getRoundLength * BLOCK_TIME)
+    monthlyReward = 2592000 // (Pos.getRoundLength * BLOCK_TIME) 
+    yearlyReward = 31536000 // (Pos.getRoundLength * BLOCK_TIME)
```

Comment:
These numbers are just how often a block is generated. Don't we need to still multiple with the block reward?

Reply:
right, probably got removed during pasting constants from calculator.. Nice that you caught that!